src/SprykerEco/Yves/AmazonPay/Controller/PaymentController.php

```diff
@@ -201,6 +198,42 @@ public function successAction(Request $request)
         ];
     }
 
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return void
+     */
+    protected function storeAmazonPaymentIntoQuote(Request $request, QuoteTransfer $quoteTransfer)
+    {
+        if ($quoteTransfer->getAmazonpayPayment() !== null) {
+            return;
+        }
+
+        $amazonPaymentTransfer = $this->buildAmazonPaymentTransfer($request);
+
+        $quoteTransfer->setAmazonpayPayment($amazonPaymentTransfer);
+        $quoteTransfer = $this->getClient()
+            ->handleCartWithAmazonPay($quoteTransfer);
+        $this->getFactory()
+            ->getQuoteClient()
+            ->setQuote($quoteTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return int
+     */
+    protected function getCurrentShipmentMethodId(QuoteTransfer $quoteTransfer)
+    {
+        if ($quoteTransfer->getShipment() === null || $quoteTransfer->getShipment()->getMethod() === null) {
+            return 0;
```

Comment:
Is it really 0? It's not a good way to receive some number as invalid id. Just return `null`